tools/report.sh

```diff
@@ -1,276 +1,296 @@
+for POD in $PODS; do
+  echo "    $POD"
+  if [[ "$POD" == *"-entity-operator-"* ]]; then
+    get_container_logs_if_present "$POD" topic-operator "$TMP"/reports/podlogs/"$POD"-topic-operator.log
+    get_container_logs_if_present "$POD" user-operator "$TMP"/reports/podlogs/"$POD"-user-operator.log
+    get_container_logs_if_present "$POD" tls-sidecar "$TMP"/reports/podlogs/"$POD"-tls-sidecar.log
+  elif [[ "$POD" =~ .*-kafka-[0-9]+ ]]; then
+    get_container_logs_if_present "$POD" kafka "$TMP"/reports/podlogs/"$POD".log
+    get_container_logs_if_present "$POD" tls-sidecar "$TMP"/reports/podlogs/"$POD"-tls-sidecar.log
+    $KUBE_CLIENT exec -i "$POD" -n "$NAMESPACE" -c kafka -- cat /tmp/strimzi.properties > "$TMP"/reports/configs/"$POD".cfg
+  elif [[ "$POD" =~ .*-zookeeper-[0-9]+ ]]; then
+    get_container_logs_if_present "$POD" zookeeper "$TMP"/reports/podlogs/"$POD".log
+    get_container_logs_if_present "$POD" tls-sidecar "$TMP"/reports/podlogs/"$POD"-tls-sidecar.log
+    $KUBE_CLIENT exec -i "$POD" -n "$NAMESPACE" -c zookeeper -- cat /tmp/zookeeper.properties > "$TMP"/reports/configs/"$POD".cfg
+  elif [[ "$POD" == *"-kafka-exporter-"* || "$POD" == *"-connect-"* || "$POD" == *"-bridge-"* || "$POD" == *"-mirror-maker-"* ]]; then
+    $KUBE_CLIENT logs "$POD" -n "$NAMESPACE" > "$TMP"/reports/podlogs/"$POD".log
+    PPLOG=$($KUBE_CLIENT logs "$POD" -p -n "$NAMESPACE" 2>/dev/null ||true)
+    if [[ -n $PPLOG ]]; then echo "$PPLOG" > "$TMP"/reports/podlogs/"$POD".log.0; fi
+  fi
+done
 
-# getting CO deployment from the pod
-$platform get deployment strimzi-cluster-operator -o yaml -n $namespace > $direct/reports/deployments/cluster-operator.yaml
-$platform get pod -l strimzi.io/kind=cluster-operator -o yaml -n $namespace > $direct/reports/pods/cluster-operator.yaml
-co_pod=$($platform get pod -l strimzi.io/kind=cluster-operator -o name -n $namespace)
-$platform logs $co_pod -n $namespace > $direct/reports/podLogs/cluster-operator.log
-$platform logs $co_pod -p -n $namespace 2>/dev/null > $direct/reports/podLogs/previous-cluster-operator.log
+CO_DEPLOY=$($KUBE_CLIENT get deploy strimzi-cluster-operator -o name -n "$NAMESPACE" --ignore-not-found) && readonly CO_DEPLOY
+if [[ -n $CO_DEPLOY ]]; then
+  echo "clusteroperator"
+  echo "    $CO_DEPLOY"
+  $KUBE_CLIENT get deploy strimzi-cluster-operator -o yaml -n "$NAMESPACE" > "$TMP"/reports/deployments/cluster-operator.yaml
+  $KUBE_CLIENT get po -l strimzi.io/kind=cluster-operator -o yaml -n "$NAMESPACE" > "$TMP"/reports/pods/cluster-operator.yaml
+  CO_POD=$($KUBE_CLIENT get po -l strimzi.io/kind=cluster-operator -o name -n "$NAMESPACE" --ignore-not-found)
+  if [[ -n $CO_POD ]]; then
+    echo "    $CO_POD"
+    CO_POD=$(echo "$CO_POD" | cut -d "/" -f 2) && readonly CO_POD
+    $KUBE_CLIENT logs "$CO_POD" -n "$NAMESPACE" > "$TMP"/reports/podlogs/cluster-operator.log
+    $KUBE_CLIENT logs "$CO_POD" -p -n "$NAMESPACE" 2>/dev/null > "$TMP"/reports/podlogs/cluster-operator.log.0
```

Comment:
Why didn't you kept the `previous-` naming? Also, unless I missed something ... why do you now capture the previous log only for the Cluster Operator and for the deployments but not for the statefulsets? (+ why does it use different mechanism in each place?)

Reply:
I used one of the common convention for naming rolled log files, with `.n` suffix (e.g. my.log, my.log.0, my.log.1, etc.). There are others, no preference at all.

Reply:
There was no log capturing for the statefulsets in the previous revision. Should I add them?

Reply:
> I used one of the common conventions for naming rolled log files, with .n suffix (e.g. my.log, my.log.0, my.log.1, etc.). There are others, no preference at all.

TBH, I'm not sure where is that common. It might be eaiser to follow what we had here as common and maintain continuity unless there is a really strong reason for changing it => for example because tool processing the output would nor handle the `previous-` prefix etc. But I do not think there is anything like that.

> What do you mean with statefulset logs? I now created a single function to get pod/container log, including previous one if present.

Sorry, with StatfefulSet logs I meant Kafka and ZooKeeper logs. They did not seemed to be captured (the previous log).

Reply:
I see previous logs in my tests, but only when there is any. I'm avoiding to create empty files. Try to delete a pod.